lib/Net/Async/Blockchain/Plugins/ETH/ERC20.pm

```diff
@@ -0,0 +1,103 @@
+package Net::Async::Blockchain::Plugins::ETH::ERC20;
+
+use strict;
+use warnings;
+no indirect;
+
+use Digest::Keccak qw(keccak_256_hex);
+use Future::AsyncAwait;
+
+our $VERSION = '0.001';
+
+use parent qw(Net::Async::Blockchain::Plugins::ETH::Utility);
+
+use constant {
+    TRANSFER_EVENT_SIGNATURE => '0x' . keccak_256_hex('Transfer(address,address,uint256)'),
+    SYMBOL_SIGNATURE         => '0x' . keccak_256_hex('symbol()'),
+    DECIMALS_SIGNATURE       => '0x' . keccak_256_hex('decimals()'),
+};
+
+=head2 enabled
+
+To disable this plugin set the return to 0
+
+=cut
+
+sub enabled { return 1 }
+
+=head2 check
+
+Check if the transactions contains ERC20 transfers
+
+=over 4
+
+=item * client L<Net::Async::Blockchain::ETH>
+
+=item * L<Net::Async::Blockchain::Transaction>
+
+=item * transaction receipt received from `eth_getTransactionReceipt`
+
+=back
+
+hashref from an array of L<Net::Async::Blockchain::Transaction>
+
+=cut
+
+async sub check {
+    my ($self, $client, $transaction, $receipt) = @_;
+    my $logs = $receipt->{logs};
+    my @transactions;
+
+    if (ref $logs eq 'ARRAY' && scalar $logs->@* > 0) {
```

Comment:
what if it's not an array? I'd just make this into an early `return unless @$logs`